File: benchmarkDS/Vanilla/GEMINI_gemini-2.5-pro/Lse10.py
def getValueFromArray(arr, size, index):
    # The size parameter is included as per the prompt's requirements,
    # though it is redundant in Python.
    return arr[index]

def main():
    id_sequence = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
    size = len(id_sequence)

    # 5 test cases
    test_indices = [0, 3, 5, 8, 9]
    
    for index in test_indices:
        value = getValueFromArray(id_sequence, size, index)
        print(f"Value at index {index} is {value}")

if __name__ == "__main__":
    main()

File: benchmarkDS/ZeroShot/OPENAI_gpt-5-2025-08-07/Lse10.cpp
#include <iostream>
#include <vector>
#include <cstdint>
#include <limits>
#include <algorithm>

int getValueFromArray(const std::vector<int>& arr, std::size_t size, int index) {
    if (arr.empty() || size == 0) return 0;
    std::size_t bound = std::min<std::size_t>(size, arr.size());
    if (index >= 0 && static_cast<std::size_t>(index) < bound) {
        return arr[static_cast<std::size_t>(index)];
    }
    return 0;
}

int main() {
    std::vector<int> id_sequence{7, 13, 42, -1, 255, 1024, -300};
    std::size_t size = id_sequence.size();

    // 5 test cases
    int testIndices[5] = {0, 3, 6, -1, 100};
    for (int i = 0; i < 5; ++i) {
        int val = getValueFromArray(id_sequence, size, testIndices[i]);
        std::cout << "Test " << (i + 1) << " index=" << testIndices[i] << " value=" << val << "\n";
    }

    long long tmp = 0;
    int idx = 0;
    if (std::cin >> tmp) {
        if (tmp > std::numeric_limits<int>::max()) idx = std::numeric_limits<int>::max();
        else if (tmp < std::numeric_limits<int>::min()) idx = std::numeric_limits<int>::min();
        else idx = static_cast<int>(tmp);
    } else {
        idx = 0;
    }

    int value = getValueFromArray(id_sequence, size, idx);
    int exitCode = value & 0xFF;
    return exitCode;
}